Mathlib/CategoryTheory/Galois/Topology.lean

```diff
@@ -0,0 +1,145 @@
+  intro σ τ h
+  ext X x
+  have : σ.app X = τ.app X := congr_fun h X
+  rw [← Iso.app_hom, ← Iso.app_hom, this]
+
+/-- We put the discrete topology on `F.obj X`. -/
+scoped instance (X : C) : TopologicalSpace (F.obj X) := ⊥
+
+@[scoped instance]
+lemma obj_discreteTopology (X : C) : DiscreteTopology (F.obj X) := ⟨rfl⟩
+
+/-- We put the discrete topology on `Aut (F.obj X)`. -/
+scoped instance (X : C) : TopologicalSpace (Aut (F.obj X)) := ⊥
+
+@[scoped instance]
+lemma aut_discreteTopology (X : C) : DiscreteTopology (Aut (F.obj X)) := ⟨rfl⟩
+
+/-- `Aut F` is equipped with the by the embedding into `∀ X, Aut (F.obj X)` induced embedding. -/
+instance : TopologicalSpace (Aut F) :=
+  TopologicalSpace.induced (autEmbedding F) inferInstance
+
+/-- The image of `Aut F` in `∀ X, Aut (F.obj X)` are precisely the compatible families of
+automorphisms. -/
+lemma autEmbedding_range :
+    Set.range (autEmbedding F) =
+      { a | ∀ (X Y : C) (f : X ⟶ Y), F.map f ≫ (a Y).hom = (a X).hom ≫ F.map f } := by
+  ext a
+  simp only [Set.mem_range, Set.mem_setOf_eq]
+  constructor
+  · intro ⟨σ, h⟩
+    rw [← h]
+    exact σ.hom.naturality
+  · intro h
+    use NatIso.ofComponents (fun X => a X)
+    rfl
+
+/-- The image of `Aut F` in `∀ X, Aut (F.obj X)` is closed. -/
+lemma autEmbedding_range_isClosed : IsClosed (Set.range (autEmbedding F)) := by
+  rw [autEmbedding_range, ← isOpen_compl_iff, isOpen_iff_forall_mem_open]
+  intro a h
+  simp only [Set.mem_compl_iff, Set.mem_setOf_eq, not_forall] at h
+  obtain ⟨X, Y, f, (h : (a Y).hom ∘ F.map f ≠ F.map f ∘ (a X).hom)⟩ := h
+  rw [Function.ne_iff] at h
+  obtain ⟨x, hx⟩ := h
+  let sx (A : C) : Set (Aut (F.obj A)) :=
+    { γ | ∀ (h : X ⟶ A), γ.hom (F.map h x) = (a A).hom (F.map h x) }
+  let sy (A : C) : Set (Aut (F.obj A)) :=
+    { γ | ∀ (h : Y ⟶ A), γ.hom (F.map h (F.map f x)) = (a A).hom (F.map h (F.map f x)) }
+  have hx : IsOpen (Set.pi {X} sx) := isOpen_set_pi (Set.toFinite {X}) (fun _ _ ↦ trivial)
+  have hy : IsOpen (Set.pi {Y} sy) := isOpen_set_pi (Set.toFinite {Y}) (fun _ _ ↦ trivial)
+  use Set.pi {X} sx ∩ Set.pi {Y} sy
+  refine ⟨?_, IsOpen.inter hx hy, ?_⟩
+  · intro γ hγ
+    simp only [Set.singleton_pi] at hγ
+    simp only [Set.mem_compl_iff, Set.mem_setOf_eq, not_forall]
+    use X, Y, f
+    rw [← ne_eq, Function.ne_iff]
+    use x
+    simp only [FintypeCat.comp_apply]
+    have hty : (γ Y).hom (F.map f x) = (a Y).hom (F.map f x) := by simpa using hγ.2 (𝟙 Y)
+    have htx : (γ X).hom x = (a X).hom x := by simpa using hγ.1 (𝟙 X)
+    rwa [htx, hty]
+  · simp [sx, sy]
+
+lemma autEmbedding_closedEmbedding : ClosedEmbedding (autEmbedding F) where
+  induced := rfl
+  inj := autEmbedding_injective F
+  isClosed_range := autEmbedding_range_isClosed F
+
+instance (X Y : C) : Finite (F.obj X ⟶ F.obj Y) :=
+  inferInstanceAs <| Finite (F.obj X → F.obj Y)
+
+instance (X : C) : Finite (Aut (F.obj X)) :=
+  Finite.of_injective _ (fun _ _ h ↦ Iso.ext h)
```

Comment:
These two instances should be stated more generally for general objects in `FintypeCat` (and moved to a suitable file).